### ISSUE TITLE
apiserver and controller manager service account private key should not use root key

### ISSUE THREAD

**Issue by justlaputa** [kelseyhightower/kubernetes-the-hard-way#230]
Hi, I found a issue regarding the private key used to do service account authentication.

on the step of [Bootstrapping the Kubernetes Control Plane](https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/master/docs/08-bootstrapping-kubernetes-controllers.md), it suggest using the root ca's private key as service account validation key, both on apiserver and controller manager:

```
ExecStart=/usr/local/bin/kube-apiserver \\
...
  --service-account-key-file=/var/lib/kubernetes/ca-key.pem \\
```

```
ExecStart=/usr/local/bin/kube-controller-manager \\
...
  --service-account-private-key-file=/var/lib/kubernetes/ca-key.pem \\
```

I think it's not good to distribute ca's private key to all master node, and use it to do service account validation key. It's better to create separate key for this, or use apiserver's own private key `/var/lib/kubernetes/kubernetes-key.pem` (which is also used to serve https)

there are some discussion here suggests use apiserver's https key:
https://github.com/kubernetes/kubernetes/issues/11000#issuecomment-120413632

and in the official gce setup script, it generates separate key for service account specifically:

https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L279

How do you think to change this part in the hard-way documents?

**Comment by tokiwinter** [comment]
I agree with this, pushing out my CAs private key to all master nodes gives me the heebie jeebies.

**Comment by tokiwinter** [comment]
FWIW, I have been implementing my k8s clusters with a dedicated private key for the service account, and all is well. My CA's private key does not exist anywhere within my Kubernetes infrastructure, _at all_. Whilst this tutorial is about simplicity, I think we need to be careful sending the message that distributing the CAs private key is okay under any circumstance. Either that, or a big bold warning somewhere on [this page](https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/master/docs/04-certificate-authority.md).

**Comment by kelseyhightower** [comment]
Fixed on the 1.10.2 branch.
